### PR TITLE
AC-5238 Add Nginx/Supervisord-level Papertrail Logging to Impact API

### PR DESCRIPTION
**Pull request by pbx**
## [AC-5238](https://masschallenge.atlassian.net/browse/AC-5238)

You can test this locally by changing [start.sh line 16](https://github.com/masschallenge/impact-api/blob/AC-5238/web/scripts/start.sh#L16) to say `"Dev"` instead of `"Prod"` -- then logs will be sent from your dev machine.

The system name on Papertrail is [impact-api](https://papertrailapp.com/systems/impact-api/events) and I've created an [Impact API](https://papertrailapp.com/groups/8659131) group in Papertrail to hold it.

### CODE REVIEW

File: web/Dockerfile [hunk 1]
```diff
@@ -3,32 +3,30 @@ FROM python:3.6
...
-
 COPY scripts/start.sh /usr/bin
-
 COPY scripts/start-nodaemon.sh /usr/bin
 
 RUN apt-get update -y
 
 # python2.7 is required to run supervisor
-RUN apt-get install -y netcat mysql-client python2.7 python-setuptools nginx
+RUN apt-get install -y gettext netcat mysql-client nginx python2.7 python-setuptools
```
Comment (mc-nathan) [inline web/Dockerfile]: Why the order change for `nginx`?  At first I thought it might be alphabetical, but then `netcat` is out of order.

**Review by mc-nathan — APPROVED** [review]
Looks reasonable.  One very minor non-blocking question.  Approved.

File: web/Dockerfile (hunk 1 above)
Comment (pbx) [reply web/Dockerfile]: Was intended to be alphabetical. Fixed.

**Review by yotammanor — DISMISSED** [review]
- I was able to see supervisord logs in papertrail, but only after I moved the `if` statement in `start.sh` to run before `service nginx restart`. Not sure why.

- Seems like nginx.conf L28 and L29 set the logs to be to stdout instead of to the relevant log files. This seems to be wrong. It is also possible that nginx logs are not configured correctly in general, since they don't output to stdout locally. Possibly a different ticket, but preferably we should just sort it out now.

A small non-blocking comment below.

**Comment by pbx** [comment]
Closing this until the remaining work (fixing the nginx log configuration issue that @yotammanor discovered) is completed.

File: web/Dockerfile [hunk 2]
```diff
@@ -3,32 +3,30 @@ FROM python:3.6
...
 COPY scripts/start-nodaemon.sh /usr/bin
 
 RUN apt-get update -y
 
 # python2.7 is required to run supervisor
-RUN apt-get install -y netcat mysql-client python2.7 python-setuptools nginx
+RUN apt-get install -y gettext mysql-client netcat nginx python-setuptools python2.7
 
+# Install remote_syslog2 to talk to papertrail
+RUN cd /tmp/ && curl -LO https://github.com/papertrail/remote_syslog2/releases/download/v0.20/remote-syslog2_0.20_amd64.deb
```
[the remark this answers is not in the document]
Comment (pbx) [reply web/Dockerfile]: Done

**Comment by pbx** [comment]
Current state: Nginx log config has been fixed, and nginx log data is being sent to papertrail. However, a number of containers created by the build process also show up in papertrail in addition to **impact-api** (they are identified by their Docker container IDs, e.g. "cf0b168fedb9". I have put some work into trying to quiet these, but have not had success yet. Still, I wanted the PR back open so the requested work can be seen.

**Comment by yotammanor** [comment]
@pbx I wasn't able to trigger an nginx log message (either from error.log or from access.log), although I did see that you got an `error` message. Any minty suggestions?

File: web/nginx/nginx.conf [hunk 3]
```diff
@@ -25,8 +25,8 @@ http {
     ##
     # Logging Settings
     ##
-    access_log /dev/stdout;
-    error_log /dev/stdout info;
+    access_log /var/log/nginx/access.log;
+    error_log /var/log/nginx/access.log info;
```
Comment (yotammanor) [inline web/nginx/nginx.conf]: this seems to be the wrong file name, assuming `/var/log/nginx/error.log` is not added by mistake.

Comment (pbx) [reply web/nginx/nginx.conf]: :headdesk:

Fixed.

**Comment by shanbady** [comment]
@yotammanor the reason you are not seeing output locally im guessing is because you are visiting http://localhost:8000 instead of https://localhost

locally - nginx is mapped to port 80 and django is mapped to 8000. if you visit http://localhost you should start seeing output in /var/log/nginx/

**Review by shanbady — APPROVED** [review]

**Pull request merged**